radix-engine-interface/src/api/wasm.rs

```diff
@@ -0,0 +1,49 @@
+use crate::api::Invocation;
+use crate::data::ScryptoDecode;
+use crate::model::CallTableInvocation;
+
+pub trait SerializableInvocation:
+    Into<CallTableInvocation> + Invocation<Output = Self::ScryptoOutput>
+{
+    type ScryptoOutput: ScryptoDecode;
+}
+
+pub type BufferId = u32;
+pub type Buffer = u64;
+
+#[macro_export]
+macro_rules! buffer_id {
```

Comment:
Should these be inline functions rather than macros instead to preserve type checking?

Reply:
There is on way to define methods on `u64`, unless we use `Buffer(u64)` which is not compitable with Rust FFI. It's the lesser evil. :)

Reply:
Maybe I can manually reproduce C layout and alignment. Let me try.

Reply:
Applied the change